qiime2/metadata/metadata.py

```diff
@@ -361,11 +356,15 @@ def __init__(self, dataframe):
 
         super().__init__(dataframe.index)
 
+        self._validate_index(dataframe.columns, axis='column')
+        dataframe.index = dataframe.index.str.strip()
```

Comment:
This actually mutates the input, which is usually something to avoid. If you look inside `_normalize_dataframe` you'll see it creates a copy to avoid changing the input.